samples/cellular/uicc_lwm2m/src/main.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <ctype.h>
+
+#include <zephyr/kernel.h>
+#include <modem/nrf_modem_lib.h>
+#include <modem/uicc_lwm2m.h>
+#include <nrf_modem_at.h>
+
+uint8_t buffer[UICC_RECORD_BUFFER_MAX];
+
+static void printk_hexdump(char *data, int len)
+{
+	static const char hexchars[16] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
+					   'a', 'b', 'c', 'd', 'e', 'f' };
+	int pos = 0;
+
+	while (pos < len) {
+		char hexbuf[256];
```

Comment:
Any chance that this could overflow since `UICC_RECORD_BUFFER_MAX` is 516 bytes?

Reply:
This is fine because here we print 16 bytes of the data on each line, making a hex dump with offset, bytes as hex and bytes as text.